cacheops/query.py

```diff
@@ -247,27 +261,72 @@ def iterator(self):
         # TODO: do not cache empty queries in Django 1.6
         superiter = self._no_monkey.iterator
         cache_this = self._cacheprofile and 'fetch' in self._cacheconf['ops']
-
+        cache_for_read = (
+            not self._cacheconf['write_only']
+            and not self._for_write
+        )
         if cache_this:
             cache_key = self._cache_key()
-            if not self._cacheconf['write_only'] and not self._for_write:
+
+            if cache_for_read:
+
                 # Trying get data from cache
                 cache_data = redis_client.get(cache_key)
+
                 if cache_data is not None:
                     results = pickle.loads(cache_data)
+
                     for obj in results:
                         yield obj
+
                     raise StopIteration
 
         # Cache miss - fallback to overriden implementation
         results = []
-        for obj in superiter(self):
+
+        # We use the lock only if we need to cache. If not needing
+        # to cache, using the lock is superfluous anyway because we
+        # *will* dog-pile by computing the queryset multiple times.
+        if cache_this and cache_for_read and USE_LOCK:
+            with redis_lock.Lock(redis_client, cache_key):
+
+                # In case the lock was aquired by more than one thread or
```

Comment:
Lock can't really be acquired by 2 processes, it's its purpose, so message sounds confusing. Probably something like "Retry fetching from cache in case previous locker filled it." will do better.